src/exportJarSteps/ExportJarTaskProvider.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {
+    CustomExecution, Event, EventEmitter, Pseudoterminal, Task, TaskDefinition, TaskProvider,
+    TaskRevealKind, TerminalDimensions, workspace,
+} from "vscode";
+import { createJarFile } from "../exportJarFileCommand";
+import { IStepMetadata } from "./IStepMetadata";
+
+export class ExportJarTaskProvider implements TaskProvider {
+
+    public static exportJarType: string = "exportjar";
+
+    public static getTask(stepMetadata: IStepMetadata): Task {
+        const targetPathSetting: string = workspace.getConfiguration("java.dependency.exportjar").get<string>("defaultTargetFolder");
+        const defaultDefinition: IExportJarTaskDefinition = {
+            type: ExportJarTaskProvider.exportJarType,
+            targetPath: targetPathSetting,
+            elements: [],
+            mainMethod: undefined,
+        };
+        const task: Task = new Task(defaultDefinition, stepMetadata.workspaceFolder, "DEFAULT_EXPORT", ExportJarTaskProvider.exportJarType,
+            new CustomExecution(async (resolvedDefinition: TaskDefinition): Promise<Pseudoterminal> => {
+                return new ExportJarTaskTerminal(resolvedDefinition, stepMetadata);
+            }));
+        task.presentationOptions.reveal = TaskRevealKind.Never;
+        return task;
+    }
+
+    public async resolveTask(_task: Task): Promise<Task> {
+        return _task;
+    }
+
+    public async provideTasks(): Promise<Task[]> {
+        return [];
+    }
+
+}
+
+class ExportJarTaskTerminal implements Pseudoterminal {
+
+    public writeEmitter = new EventEmitter<string>();
+    public closeEmitter = new EventEmitter<void>();
+
+    public onDidWrite: Event<string> = this.writeEmitter.event;
```

Comment:
Nowhere so far. Will remove useless fileds.

Reply:
They are the members of the interface `Pseudoterminal`, for `onDidWrite`, it is a required field. For `onDidClose`, it will be called when the task exits.